src/main/java/com/limechain/sync/warpsync/SyncedState.java

```diff
@@ -0,0 +1,64 @@
+package com.limechain.sync.warpsync;
+
+import com.limechain.chain.lightsyncstate.Authority;
+import com.limechain.constants.GenesisBlockHash;
+import com.limechain.network.Network;
+import com.limechain.network.protocol.blockannounce.scale.BlockAnnounceHandshake;
+import com.limechain.rpc.http.server.AppBean;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.math.BigInteger;
+
+public class SyncedState {
+    private static final SyncedState INSTANCE = new SyncedState();
+    @Getter
+    @Setter
+    public Hash256 lastFinalizedBlockHash;
```

Comment:
I think it is a better practice to declare variables as `private` and use the getters/setters to access them.